library/Translate/src/Juvix/Contextify/Environment.hs

```diff
@@ -150,18 +176,38 @@ class
     (NameSymbol.T -> Bool) ->
     contextTo m Sexp.T Sexp.T Sexp.T
 
+instance ToContextTraversal PassManual Context.ContextForms where
+  promotePass (PassManual pass) trigger =
+    let f form ctx =
+          Sexp.foldSearchPredManualRecurse
+            form
+            (trigger, pass ctx)
+            -- Hack? We unbind the primtiive form for being a
+            -- binder. Why was this a binder in the first place? If we
+            -- change it we get errors throughout a resolution step.
```

Comment:
Is this a question that you've been able to answer by now?  I'm not exactly sure what this comment means or what the implications are, but if it's unclear why the code didn't work in some other form in which you expected it to work, then I'd worry that there would still be a bug that could have other consequences that we're just working around here.

Reply:
Yeah I remembered last night. Due to having no quotes, we have to do resolution on all symbols, thus when we see `(:primitive foo)` we need to say it's a binder as otherwise we will try to resolve foo to a module that we've opened

Reply:
Cool, if you update the comment to reflect that, that'll be helpful.

Good stuff!  Thanks.